dockerfile_parse/parser.py

```diff
@@ -266,44 +266,101 @@ def json(self):
         insndescs = [{insndesc['instruction']: insndesc['value']} for insndesc in self.structure]
         return json.dumps(insndescs)
 
+    @property
+    def parent_images(self):
+        """
+        :return: list of parent images -- one image per each stage's FROM instruction
+        """
+        parents = []
+        for instr in self.structure:
+            if instr['instruction'] != 'FROM':
+                continue
+            image, _ = image_from(instr['value'])
+            if image is not None:
```

Comment:
I don't think we have testing that covers this not being the case.

Reply:
Added one. And actually setting parent_images would be broken in that case. Well, everything would be broken anyway, but we can at least do no further harm.